File: app/src/test/java/org/blitzortung/android/util/MeasurementSystemTest.java
package org.blitzortung.android.util;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;

import static org.hamcrest.core.Is.is;
import static org.junit.Assert.assertThat;

@RunWith(RobolectricTestRunner.class)
public class MeasurementSystemTest {

    @Test
    public void testMetricFactor() {
        float metricDistance = MeasurementSystem.METRIC.calculateDistance(123456);
        assertThat(metricDistance, is(123.456f));
    }

    @Test
    public void testMetricUnitName() {
        assertThat(MeasurementSystem.METRIC.getUnitName(), is("km"));
    }

    @Test
    public void testImperialFactor() {
        float metricDistance = MeasurementSystem.IMPERIAL.calculateDistance(2 * 1609.344f);
        assertThat(metricDistance, is(2f));
    }

    @Test
    public void testImperialUnitName() {
        assertThat(MeasurementSystem.IMPERIAL.getUnitName(), is("mi."));
    }
}

File: app/src/test/java/org/blitzortung/android/util/MeasurementSystemTest.kt
package org.blitzortung.android.util

import org.assertj.core.api.Assertions.assertThat
import org.blitzortung.android.app.R
import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.RobolectricTestRunner
import org.robolectric.annotation.Config

@RunWith(RobolectricTestRunner::class)
@Config(manifest = Config.NONE)
class MeasurementSystemTest {
    @Test
    fun testMetricFactor() {
        val metricDistance = MeasurementSystem.METRIC.calculateDistance(123456f)
        assertThat(metricDistance).isEqualTo(123.456f)
    }

    @Test
    fun testMetricUnitName() {
        assertThat(MeasurementSystem.METRIC.unitNameString).isEqualTo(R.string.unit_km)
    }

    @Test
    fun testImperialFactor() {
        val metricDistance = MeasurementSystem.IMPERIAL.calculateDistance(2 * 1609.344f)
        assertThat(metricDistance).isEqualTo(2f)
    }

    @Test
    fun testImperialUnitName() {
        assertThat(MeasurementSystem.IMPERIAL.unitNameString).isEqualTo(R.string.unit_miles)
    }
}
